lib/docker-toolbelt.coffee

```diff
@@ -304,3 +304,16 @@ Docker::getRegistryAndName = Promise.method (image) ->
 	[ ..., registry = 'docker.io', imageName, tagName = 'latest' ] = match
 	throw new Error('Invalid image name, expected domain.tld/repo/image format.') if not imageName
 	return { registry, imageName, tagName }
+
+# Given an object representing a docker image, in the same format as given
+# by getRegistryAndName, compile it back into a docker image string, which
+# can be used in Docker command etc
+# Example: { registry: "registry.resinstaging.io", imageName: "resin/rpi", tag: "1234"}
+#		=> registry.resinstaging.io/resin/rpi:1234
+Docker::compileRegistryAndName = Promise.method (image) ->
+	registry =
+		if image.registry isnt ''
+			image.registry + '/'
+		else
+			''
+	return "#{registry}#{image.imageName}:#{image.tagName}"
```

Comment:
The comments say the tag is the `tag` attribute of `image`, but here you use `tagName`